In [89]:
import pandas as pd

**Inability to face unexpected financial expenses**

---

Percent of people unable to face unpredicted financial expenses
https://ec.europa.eu/eurostat/databrowser/view/ILC_MDES04__custom_70943/bookmark/table?lang=en&bookmarkId=8e195f40-5d4b-40db-9a11-49a57cef26f5

In [90]:
df = pd.read_csv('../../data/inability_to_face_unexpected_expenses_(wealth).csv')

In [91]:
df.head()

DATAFLOW        LAST UPDATE    freq  \
0  ESTAT:ILC_MDES04$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual   
1  ESTAT:ILC_MDES04$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual   
2  ESTAT:ILC_MDES04$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual   
3  ESTAT:ILC_MDES04$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual   
4  ESTAT:ILC_MDES04$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual   

           hhtyp incgrp        unit      geo  TIME_PERIOD  OBS_VALUE OBS_FLAG  \
0  Single person  Total  Percentage  Albania         2017       71.5      NaN   
1  Single person  Total  Percentage  Albania         2018       66.5      NaN   
2  Single person  Total  Percentage  Albania         2019       65.1      NaN   
3  Single person  Total  Percentage  Albania         2020       58.6      NaN   
4  Single person  Total  Percentage  Albania         2021       58.3      NaN   

   CONF_STATUS  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

In [92]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1910 entries, 0 to 1909
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   DATAFLOW     1910 non-null   object 
 1   LAST UPDATE  1910 non-null   object 
 2   freq         1910 non-null   object 
 3   hhtyp        1910 non-null   object 
 4   incgrp       1910 non-null   object 
 5   unit         1910 non-null   object 
 6   geo          1910 non-null   object 
 7   TIME_PERIOD  1910 non-null   int64  
 8   OBS_VALUE    1910 non-null   float64
 9   OBS_FLAG     155 non-null    object 
 10  CONF_STATUS  0 non-null      float64
dtypes: float64(2), int64(1), object(8)
memory usage: 164.3+ KB


OBS_FLAG = 155 meaning 155 out of 1910 values were either estimated or a break in the time series has occured

In [93]:
columns_to_drop = ["OBS_FLAG", "DATAFLOW", "CONF_STATUS", "freq", "incgrp", "unit", "LAST UPDATE"]
df = df.drop(columns=[col for col in columns_to_drop if col in df.columns])

In [94]:
df = df.dropna()

In [95]:
for col in df.columns:
    print(f"Uniques values in {col}: {df[col].unique()}")

Uniques values in hhtyp: ['Single person' 'Single person with dependent children'
 'One adult 65 years or over' 'One adult younger than 65 years' 'Total']
Uniques values in geo: ['Albania' 'Austria' 'Belgium' 'Bulgaria' 'Switzerland' 'Cyprus' 'Czechia'
 'Germany' 'Denmark'
 'Euro area (EA11-1999, EA12-2001, EA13-2007, EA15-2008, EA16-2009, EA17-2011, EA18-2014, EA19-2015, EA20-2023)'
 'Euro area - 17 countries (2011-2013)' 'Euro area - 18 countries (2014)'
 'Euro area - 19 countries  (2015-2022)'
 'Euro area – 20 countries (from 2023)' 'Estonia' 'Greece' 'Spain'
 'European Union (EU6-1958, EU9-1973, EU10-1981, EU12-1986, EU15-1995, EU25-2004, EU27-2007, EU28-2013, EU27-2020)'
 'European Union - 15 countries (1995-2004)'
 'European Union - 27 countries (2007-2013)'
 'European Union - 27 countries (from 2020)'
 'European Union - 28 countries (2013-2020)' 'Finland' 'France' 'Croatia'
 'Hungary' 'Ireland' 'Iceland' 'Italy' 'Lithuania' 'Luxembourg' 'Latvia'
 'Montenegro' 'North Macedonia' '

In [96]:
df = df[~df["geo"].isin(['European Union - 27 countries (from 2020)',
                         'Euro area (EA11-1999, EA12-2001, EA13-2007, EA15-2008, EA16-2009, EA17-2011, EA18-2014, EA19-2015, EA20-2023)',
                         'Euro area - 17 countries (2011-2013)',
                         'European Union (EU6-1958, EU9-1973, EU10-1981, EU12-1986, EU15-1995, EU25-2004, EU27-2007, EU28-2013, EU27-2020)',
                         'European Union - 15 countries (1995-2004)',
                         'European Union - 27 countries (2007-2013)',
                         'European Union - 27 countries (from 2020)',
                         'European Union - 28 countries (2013-2020)'
                         'Euro area - 19 countries  (2015-2022)',
                         'Euro area – 20 countries (from 2023)',
                         'Euro area - 18 countries (2014)',
                         'Euro area - 19 countries  (2015-2022)',
                         'European Union - 28 countries (2013-2020)'
                         ])]

columns_to_drop = ["freq", "indic_de", "statinfo", "age", "unit", 'isced11']
df = df.drop(columns=[col for col in columns_to_drop if col in df.columns])

In [97]:
for col in df.columns:
    print(f"Uniques values in {col}: {df[col].unique()}")

Uniques values in hhtyp: ['Single person' 'Single person with dependent children'
 'One adult 65 years or over' 'One adult younger than 65 years' 'Total']
Uniques values in geo: ['Albania' 'Austria' 'Belgium' 'Bulgaria' 'Switzerland' 'Cyprus' 'Czechia'
 'Germany' 'Denmark' 'Estonia' 'Greece' 'Spain' 'Finland' 'France'
 'Croatia' 'Hungary' 'Ireland' 'Iceland' 'Italy' 'Lithuania' 'Luxembourg'
 'Latvia' 'Montenegro' 'North Macedonia' 'Malta' 'Netherlands' 'Norway'
 'Poland' 'Portugal' 'Romania' 'Serbia' 'Sweden' 'Slovenia' 'Slovakia'
 'Türkiye' 'United Kingdom' 'Kosovo*']
Uniques values in TIME_PERIOD: [2017 2018 2019 2020 2021 2015 2016 2022 2023 2024]
Uniques values in OBS_VALUE: [71.5 66.5 65.1 58.6 58.3 30.  31.  30.1 28.5 26.1 23.8 23.1 26.8 30.2
 33.5 33.6 33.8 33.2 31.8 31.2 32.2 29.6 28.2 80.9 80.  78.4 74.8 72.9
 71.3 64.2 61.3 64.8 60.4 23.7 22.8 22.5 23.2 22.9 22.3 22.1 21.1 20.6
 63.4 61.6 52.8 47.8 46.8 41.4 40.2 43.2 42.4 49.5 45.5 42.7 36.  30.5
 28.  28.6 29.3 26.4 43.8 42

In [98]:
df = df.rename(columns={"OBS_VALUE": "Percent of people affected", "TIME_PERIOD": "Time period"})

In [99]:
df.to_csv("../../processed_data/inability_to_face_unexpected_expenses_(wealth).csv", index=False)